### DISCUSSION TITLE
How to reconnect to server

### DISCUSSION THREAD

**Discussion by anhnhoktvn** [socketio/socket.io#4920] [Q&A]
For a long time connection, I got 
```
failed: Error in connection establishment: net::ERR_NAME_NOT_RESOLVED
doOpen @ index-fb0b8ccc.js:1
```
I'd like to reconnect to the server to keep the data  of the connection (ex: socket.data)
With 
```
    "socket.io": "^4.7.2",
    "socket.io-client": "^4.7.2",
```
How could I do that? 
I have tried 
```
socket.connect()
```
But it doesn't work
I found some answers using 
```
socket.socket.connect() 
``
But it still doesn't work

**Comment by darrachequesne** [comment]
"ERR_NAME_NOT_RESOLVED" usually means the client is not connected to the Internet (so the DNS request fails). In that case, I'm afraid that there is nothing we can do.

**Reply by anhnhoktvn** [reply]
I understand that it's a problem with network. The thing is some clients are using not very good network. 
I found that I can wait a short time, disconnect and reconnect (for force new connection). But do that will lost the socket.data because the server see that a new connection. I hope it can somehow keep the data in this case.
Externally, I may move that data out of socket.data and recheck when new connection has handshake.old_id (something like that) and replace old socket with new socket. But I guess the library has something similar to that.

**Reply by darrachequesne** [reply]
In that case, please take a look at the [Connection state recovery feature](https://socket.io/docs/v4/connection-state-recovery).

**Reply by anhnhoktvn** [reply]
I think I'm using this feature. It just didn't work. if you have to force new connection, it can not recover. if you not force new connection. the connection broken by ERR_NAME_NOT_RESOLVED and it won't reconnect again.

**Reply by darrachequesne** [reply]
That's weird, it should work. Could you please provide a minimal example reproducing the issue, so we can help you debugging what's wrong? You can start from https://github.com/socketio/socket.io-fiddle.

**Comment by LyraHarring** [comment]
The “[ERR_NAME_NOT_RESOLVED](https://certera.com/kb/how-to-resolve-err_name_not_resolved-error/)” error message pops up when your browser cannot translate the website’s domain name into an IP address. This may be a result of DNS server errors, network complications, or browser setting problems, among others. There are many different steps that will help to fix this problem and bring comfortable browsing back.
